workflow/react-first-steps/src/components/RandomCocktail/index.js

```diff
@@ -1,18 +1,91 @@
-import {Grid, DialogContent, Typography} from '@material-ui/core';
+import { useEffect } from 'react'
+import { Grid, DialogContent, Typography, Tooltip, IconButton} from '@material-ui/core'
+import { useSelector, useDispatch } from "react-redux"
+import { bindActionCreators } from 'redux'
+import { actionCreators } from '../../store/index'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faStar as farStar } from '@fortawesome/free-regular-svg-icons'
+import Loading from '../Loading'
+import useStyles from './style.js'
+
+var _ = require('lodash');
 
 const RandomCocktail = () => {
-    return (
-      <Grid container direction="column">
-        <DialogContent>
-          <Typography variant = "h4" align="center">
-            Cocktail Name
+  const classes = useStyles()
+
+  const state = useSelector((state) => state)
+
+  const dispatch = useDispatch()
+  const {fetchRandomCocktail} = bindActionCreators(actionCreators, dispatch)
+
+  useEffect(()=>{
+    fetchRandomCocktail()
+  },[])
+
+  const getIngredients = () => {
+    const result = []
+    for (let i = 1; i < 16; i++){
+      const ingredient = _.get(state, 'fetchRandomCocktail.cocktail.drinks[0].strIngredient' + i)
+      const measure = _.get(state, 'fetchRandomCocktail.cocktail.drinks[0].strMeasure' + i)
+      if (ingredient !== null){
+        result.push(
+          <tr key = {i}>
+            <td>{i}</td>
+            <td className={classes.ingredient}>{ingredient}</td>
+            <td>{measure}</td>
+          </tr>
+        )
+      } else break;
+    }
+    return result
+  }
+
+  if (state.fetchRandomCocktail.loading) {return <Loading/>} 
+  else return (
+    <Grid container direction="column" align="center">
+      <DialogContent>
+        <Grid item align="center">
+          <Typography variant = "h4">
+            {state.fetchRandomCocktail.cocktail.drinks[0].strDrink}
```

Comment:
add variable for state.fetchRandomCocktail.cocktail.drinks[0].strDrinkThumb
this code is too heavy